partiql-eval/src/main/java/org/partiql/eval/compiler/Strategy.java

```diff
@@ -34,8 +34,20 @@ public Pattern getPattern() {
      * Applies the strategy to a logical plan operator and returns the physical operation (expr).
      *
      * @param match holds the matched operators
+     * @param callback for compiling arguments of matched operators
      * @return the physical operation
      */
     @NotNull
-    public abstract Expr apply(@NotNull Match match);
+    public abstract Expr apply(@NotNull Match match, @NotNull Callback callback);
```

Comment:
Mentioned this in another comment, but if we hope to provide default strategies for some nodes (such as `RelScan`), we need to also pass the execution mode.